mllib-local/src/test/scala/org/apache/spark/ml/linalg/VectorsSuite.scala

```diff
@@ -72,6 +72,12 @@ class VectorsSuite extends SparkMLFunSuite {
     }
   }
 
+  test("sparse vector construction with negative indices") {
+    intercept[IllegalArgumentException] {
+      Vectors.sparse(3, Array(-1, 1), Array(3.0, 5.0))
+    }
+  }
+
```

Comment:
Lastly, I believe, do we have a test that exercises the check for missorted indices? I don't see one; I suppose one should have been added before.

Same for Python.

I'm OK with the error messages, but now seeing that Python has parallel checks, it'd be nice to make the error messages match.